onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.h

```diff
@@ -39,14 +40,14 @@ class ModelBuilder {
     CPU_ONLY,      // use CPU only
   };
 
-  ModelBuilder(const GraphViewer& graph_viewer);
+  ModelBuilder(std::unique_ptr<Model> model, const GraphViewer& graph_viewer);
 
```

Comment:
ModelBuilder doesn't need to take ownership here, so pass Model as a reference and not as a unique_ptr. 

Also Compile doesn't need to take a unique_ptr<Model> - it can just update the Model reference it was constructed with.